caliban_toolbox/log_file.py

```diff
@@ -0,0 +1,244 @@
+# All rights reserved.
+#
+# Licensed under a modified Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.github.com/vanvalenlab/deepcell-toolbox/LICENSE
+#
+# The Work provided may be used for non-commercial academic purposes only.
+# For any other use of the Work, including commercial use, please contact:
+# vanvalenlab@gmail.com
+#
+# Neither the name of Caltech nor the names of its contributors may be used
+# to endorse or promote products derived from this software without specific
+# prior written permission.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import re
+import json
+
+import os
+import stat
+import pandas as pd
+import numpy as np
+import requests
+
+from caliban_toolbox.aws_functions import connect_aws, aws_transfer_file
+
+
+def create_upload_log(base_dir, stage, aws_folder, filenames, filepaths, job_id,
+                      pixel_only=False, label_only=False, rgb_mode=False):
+    """Generates a csv log of parameters used during job creation for subsequent use in pipeline.
+
+    Args:
+        base_dir: full path to directory where job results will be stored
+        stage: specifies stage in pipeline for jobs requiring multiple rounds of annotation
+        aws_folder: folder in aws bucket where files be stored
+        filenames: list of all files to be uploaded
+        filepaths: list of complete urls to images in Amazon S3 bucket
+        job_id: internal Figure Eight id for job
+        pixel_only: flag specifying whether annotators will be restricted to pixel edit mode
+        label_only: flag specifying whether annotators will be restricted to label edit mode
+        rgb_mode: flag specifying whether annotators will view images in RGB mode
+    """
+
+    data = {'project_url': filepaths,
+            'filename': filenames,
+            'stage': stage,
+            'aws_folder': aws_folder,
+            'job_id': job_id,
+            'pixel_only': pixel_only,
+            'label_only': label_only,
+            'rgb_mode': rgb_mode}
+    dataframe = pd.DataFrame(data=data, index=range(len(filepaths)))
+
+    # create file location, name file
+    log_dir = os.path.join(base_dir, 'logs')
+    if not os.path.isdir(log_dir):
+        os.makedirs(log_dir)
+
+        # add folder modification permissions to deal with files from file explorer
+        mode = stat.S_IRWXO | stat.S_IRWXU | stat.S_IRWXG
+        os.chmod(log_dir, mode)
+
+    # save csv file
+    dataframe.to_csv(os.path.join(log_dir, 'upload_log.csv'), index=False)
```

Comment:
Totally agree about overwriting. Currently it will overwrite. Next PR when I add in the multiple_stages script I'm going to enforce sequential CSV names and check that file doesn't exist. I just haven't worked out the exact syntax yet.

Reply:
I wasn't sure what to call this. Something more informative would be great that signifies this file contains the necessary information for creating a figure8 job.